mods-unpacked/ITR-SmarterDealer/bruteforce.gd

```diff
@@ -795,96 +704,85 @@ static func GetBestChoiceAndDamage_Internal(roundType: int, liveCount: int, blan
-			if not tempStates.usedHandsaw:
-				resultIfSelfShootLive = GetBestChoiceAndDamage_Internal(roundType, liveCount - originalRemove, blankCount - invertedRemove, liveCount_max, opponent, player.use("health", damageToDeal), TempStates.new())
-			resultIfShootLife = resultIfShootLife.clone()
-			resultIfSelfShootLive = resultIfSelfShootLive.clone()
+			resultIfShootLive = GetBestChoiceAndDamage_Internal(roundType, liveCount - originalRemove, blankCount - invertedRemove, liveCount_max, opponent.use("health", damageToDeal), player, TempStates.new())
+			resultIfSelfShootLive = GetBestChoiceAndDamage_Internal(roundType, liveCount - originalRemove, blankCount - invertedRemove, liveCount_max, opponent, player.use("health", damageToDeal), TempStates.new())
 		else:
 			var newTempState := TempStates.new()
 			newTempState.handcuffState = HANDCUFF_FREENEXT
-			resultIfShootLife = GetBestChoiceAndDamage_Internal(roundType, liveCount - originalRemove, blankCount - invertedRemove, liveCount_max, player, opponent.use("health", damageToDeal), newTempState)
-			if not tempStates.usedHandsaw:
-				resultIfSelfShootLive = GetBestChoiceAndDamage_Internal(roundType, liveCount - originalRemove, blankCount - invertedRemove, liveCount_max, player.use("health", damageToDeal), opponent, newTempState)
-
-		options[OPTION_SHOOT_OTHER].mutAdd(resultIfShootLife.mult(liveChance))
-
-		if not tempStates.usedHandsaw:
-			options[OPTION_SHOOT_SELF].mutAdd(resultIfSelfShootLive.mult(liveChance))
+			resultIfShootLive = GetBestChoiceAndDamage_Internal(roundType, liveCount - originalRemove, blankCount - invertedRemove, liveCount_max, player, opponent.use("health", damageToDeal), newTempState.clone())
+			resultIfSelfShootLive = GetBestChoiceAndDamage_Internal(roundType, liveCount - originalRemove, blankCount - invertedRemove, liveCount_max, player.use("health", damageToDeal), opponent, newTempState)
 
-		if itemFrom.beer > 0:
-			var beerResult = GetBestChoiceAndDamage_Internal(roundType, liveCount - originalRemove, blankCount - invertedRemove, liveCount_max, beerPlayer, beerOpponent, tempStates.SkipBullet())
-			options[OPTION_BEER].mutAdd(beerResult.mult(liveChance))
+		options[OPTION_SHOOT_OTHER].mutAdd(resultIfShootLive.mult(liveChance))
+		options[OPTION_SHOOT_SELF].mutAdd(resultIfSelfShootLive.mult(liveChance))
 
 	if blankChance > 0:
-		if not tempStates.usedHandsaw:
-			var resultIfSelfShootBlank := GetBestChoiceAndDamage_Internal(roundType, liveCount - invertedRemove, blankCount - originalRemove, liveCount_max, player, opponent, tempStates.SkipBullet())
-			options[OPTION_SHOOT_SELF].mutAdd(resultIfSelfShootBlank.mult(blankChance))
+		var selfShootState := tempStates.SkipBullet()
+		selfShootState.usedHandsaw = false
+		var resultIfSelfShootBlank := GetBestChoiceAndDamage_Internal(roundType, liveCount - invertedRemove, blankCount - originalRemove, liveCount_max, player, opponent, selfShootState)
```

Comment:
The way the game handles shooting yourself when is different for the player and the dealer, did you fix this?

Reply:
Yes. There's some extra logic in DealerChoice and EndDealerTurn in DealerIntelligence.gd to make it behave the same for the dealer as for the player.